### PR TITLE
Add documentation of recommended settings and dependencies

### PR DESCRIPTION
**Pull request by markxnelson**
This PR adds documentation for the recommended project structure, dependencies and application settings for various scenarios, to take advantage of the built-in platform services.

### CODE REVIEW

**Review by andytael — DISMISSED** [review]

File: docs-source/spring/content/development/_index.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+This section provides information about how to develop and deploy Spring Boot applications
+with the Oracle Backend for Spring Boot and Microservices. 
+
+Spring Boot applications can be developed the normal way, with no special requirements, and
```
Comment (rhondaday) [inline docs-source/spring/content/development/_index.md:4]: Change to "Spring Boot applications can be developed with no special requirements and can be deployed ..."

File: docs-source/spring/content/development/_index.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+be deployed into Oracle Backend for Spring Boot and Microservices.  However, if you do opt-in to the platform
```
Comment (rhondaday) [inline docs-source/spring/content/development/_index.md:5]: The sentence "However, if you ..." doesn't make sense. Should "provides" be changed to "provided"?

File: docs-source/spring/content/development/_index.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+services provides and the CLI, you can shorten your development time and avoid unnecessary
+work.
+
+Oracle Backend for Spring Boot provides the following services that applications can choose
```
Comment (rhondaday) [inline docs-source/spring/content/development/_index.md:9]: Delete "choose to"

File: docs-source/spring/content/development/_index.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+to use: 
+
+- An Oracle Autonomous Database instance in which applications can manage relational,
+  document, spatial, graph and other types of data, use Transactional Event Queues for
+  messaging and events using JMS, Kafka or REST interfaces, and even run ML models.
```
Comment (rhondaday) [inline docs-source/spring/content/development/_index.md:14]: expand "ML"

File: docs-source/spring/content/development/_index.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+- A Kubernetes cluster that applications can run in, with namespaces pre-configured with
```
Comment (rhondaday) [inline docs-source/spring/content/development/_index.md:15]: "A Kubernetes cluster in which application can run, with ..."

File: docs-source/spring/content/development/_index.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  Kubernetes Secrets and ConfigMaps for access to the Oracle Autonomous Database instance
+  associated with the backend.
+- An APISIX API Gateway that can be used to expose service endpoints outside the Kubernetes
```
Comment (rhondaday) [inline docs-source/spring/content/development/_index.md:18]: use "Apache APISIX"

File: docs-source/spring/content/development/_index.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  cluster, to the public internet.  All standard APISIX features like traffic management, 
+  monitoring, authentication, and so on, are available for use.
+- Spring Eureka Service Registry for service discovery.  The API Gateway and monitoring
+  services are pre-configured to use this registry for service discovery.
+- Spring Config Server to serve externalized configuration information to applications.
```
Comment (rhondaday) [inline docs-source/spring/content/development/_index.md:23]: use "Spring Cloud Config server"

File: docs-source/spring/content/development/_index.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  This is configured to store the configuration data in the Oracle Autonomous Database
+  instance associated with the backend.
+- Netflix Conductor OSS for running workflows to orchestrate your services.
+- Hashicorp Vault for storing sensitive information.
+- Spring Admin for monitoring your services.
```
Comment (rhondaday) [inline docs-source/spring/content/development/_index.md:28]: Should it be "Spring Boot Admin"?

File: docs-source/spring/content/development/_index.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+- Prometheus and Grafana for collecting and visualizing metrics and for alerting. 
+- Jaeger for distributed tracing.  Applictions deployed with the Oracle Backend for
```
Comment (rhondaday) [inline docs-source/spring/content/development/_index.md:30]: typo: Applications

File: docs-source/spring/content/development/_index.md [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+  Spring Boot CLI will have the Open Telemetry collector automatically added as a Java
```
Comment (rhondaday) [inline docs-source/spring/content/development/_index.md:31]: use "OpenTelemetry Collector"

File: docs-source/spring/content/development/_index.md [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+  agent to provide tracing from the application into the Oracle Database. 
+
+An integrated development environment is recommended for developing applications. Oracle
+recommend Visual Studio Code or IntelliJ.
```
Comment (rhondaday) [inline docs-source/spring/content/development/_index.md:35]: recommend -> recommends

File: docs-source/spring/content/development/project.md [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+            <scope>import</scope>
+        </dependency>
+    </dependencies>
+</dependencyManagement>    
+```
+
+### Spring Application Configuration
+
+Oracle recommends the following configuration to configure the application access to
+use the built-in services, including the Spring Eureka Service Registry and the
```
Comment (rhondaday) [inline docs-source/spring/content/development/project.md:69]: use "Spring Boot Eureka service registry"

File: docs-source/spring/content/development/project.md [hunk 13]
```diff
@@ -0,0 +1,205 @@
...
+  endpoints:
+    web:
+      exposure:
+        include: "*"
+  metrics:
+    tags:
+      application: ${spring.application.name}
+```
+
+The variables in this configuration will be automatically injected to your deployment
```
Comment (rhondaday) [inline docs-source/spring/content/development/project.md:103]: will be -> are

File: docs-source/spring/content/development/project.md [hunk 14]
```diff
@@ -0,0 +1,205 @@
...
+and pods when you use the Oracle Backend for Spring Boot and Microservices CLI to deploy your application. 
+
+#### Data Sources
+
+If your application uses a data source, you should add the following configuration.  Note that this
```
Comment (rhondaday) [inline docs-source/spring/content/development/project.md:108]: "If your ... source, then add the ..."

File: docs-source/spring/content/development/project.md [hunk 15]
```diff
@@ -0,0 +1,205 @@
...
+example shows JPA, if you are using JDBC you should use the appropriate configuration.
```
Comment (rhondaday) [inline docs-source/spring/content/development/project.md:109]: "Note that this example shows the Java Persistence API (JPA)". If you are using JDBC, then use the appropriate configuration:"

File: docs-source/spring/content/development/project.md [hunk 16]
```diff
@@ -0,0 +1,205 @@
...
+    type: oracle.ucp.jdbc.PoolDataSource
+    oracleucp:
+      connection-factory-class-name: oracle.jdbc.pool.OracleDataSource
+      connection-pool-name: AccountConnectionPool
+      initial-pool-size: 15
+      min-pool-size: 10
+      max-pool-size: 30
+```
+
+The variables in this configuration will be automatically injected to your deployment
```
Comment (rhondaday) [inline docs-source/spring/content/development/project.md:136]: will be -> are

File: docs-source/spring/content/development/project.md [hunk 17]
```diff
@@ -0,0 +1,205 @@
...
+spring:  
+  liquibase:
+    change-log: classpath:db/changelog/controller.yaml
+    url: ${spring.datasource.url}
+    user: ${liquibase.datasource.username}
+    password: ${liquibase.datasource.password}
+    enabled: ${LIQUIBASE_ENABLED:true}
+```
+
+The variables in this configuration will be automatically injected to your deployment
```
Comment (rhondaday) [inline docs-source/spring/content/development/project.md:170]: will be -> are

File: docs-source/spring/content/development/project.md [hunk 18]
```diff
@@ -0,0 +1,205 @@
...
+and pods when you use the Oracle Backend for Spring Boot and Microservices CLI to deploy your application. 
+When you use the `deploy` command, you must specify the `liquibase-db` parameter and
+provide a user with sufficient privileges, generally this will be to create and alter
```
Comment (rhondaday) [inline docs-source/spring/content/development/project.md:173]: "... this creates and alters users and grants roles and privileges."

File: docs-source/spring/content/development/project.md [hunk 19]
```diff
@@ -0,0 +1,205 @@
...
+users and to grant roles and privileges.  If your service uses JMS, this use may also
```
Comment (rhondaday) [inline docs-source/spring/content/development/project.md:174]: "If your services uses Java Message Service (JMS), then you may also need execute ... "

File: docs-source/spring/content/development/project.md [hunk 20]
```diff
@@ -0,0 +1,205 @@
...
+need execute permissions on `dbms.aq_adm`, `dbms.aq_in` and `dbms.aq_jms`.
+
+
+#### Oracle Transaction Manager for Microservices
+
+If you are using Oracle Transaction Manager for Microservices (MicroTx) to manage
+data consistency across microservices' data stores, you should add the following
```
Comment (rhondaday) [inline docs-source/spring/content/development/project.md:181]: "... microservice data stores, then you ..."

File: docs-source/spring/content/development/setup.md [hunk 21]
```diff
@@ -0,0 +1,128 @@
+
+This page provides details on how to set up your development environment to work with
+Oracle Backend for Spring Boot and Microservices. 
+
+The following platforms are recommended for a development environment:
+
+- Windows 10 or 11, preferrably with Windows Subsystem for Linux 2
```
Comment (rhondaday) [inline docs-source/spring/content/development/setup.md:7]: use "Microsoft Windows"

File: docs-source/spring/content/development/setup.md [hunk 22]
```diff
@@ -0,0 +1,128 @@
...
+- macOS (11 or later recommended) on Intel or Apple silicon
+- Linux, for example Oracle Linux, Ubuntu, etc.
```
Comment (rhondaday) [inline docs-source/spring/content/development/setup.md:9]: etc. -> and so on.

File: docs-source/spring/content/development/setup.md [hunk 23]
```diff
@@ -0,0 +1,128 @@
...
+The following tools are recommended for a development environment:
+
+- Integrated Development Environment, for example Visual Studio Code
+- Java Development Kit, for example Oracle, OpenJDK, or GraalVM 
+- Maven or Gradle for build and testing automation
+
+If you wish to test locally or offline, the following additional tools are recommended:
+
+- A container platform, for example Rancher Desktop
+- An Oracle Database (in a container)
```
Comment (rhondaday) [inline docs-source/spring/content/development/setup.md:20]: Database -> database

File: docs-source/spring/content/development/setup.md [hunk 24]
```diff
@@ -0,0 +1,128 @@
...
+3.0 requires at least Java 17.  If you want to use JVM Virtual Threads, then Java 21 is recommended.
+
+**Note**: If you are using Spring Boot 2.x, Oracle encourages you to use at least Java 17, unless
+you have a specific reason to stay on Java 11. Refer to the [Spring Boot Support](https://spring.io/projects/spring-boot#support)
+page for information on support dates for Spring Boot 2.x.
+
+You can download the latest x64 Java 17 Development Kit from
+[this permalink](https://download.oracle.com/java/17/latest/jdk-17_linux-x64_bin.tar.gz).
+
+Decompress the archive in your chosen location, for example your home directory and then add it to your path:
```
Comment (rhondaday) [inline docs-source/spring/content/development/setup.md:47]: "Decompress ... location (for example, your home directory) and ..."

File: docs-source/spring/content/development/setup.md [hunk 25]
```diff
@@ -0,0 +1,128 @@
...
+You can verify it is installed with this command:
+
+```bash
+$ java -version
+java version "17.0.3" 2022-04-19 LTS
+Java(TM) SE Runtime Environment (build 17.0.3+8-LTS-111)
+Java HotSpot(TM) 64-Bit Server VM (build 17.0.3+8-LTS-111, mixed mode, sharing)
+```
+
+**Note: Native Images:** If you want to compile your Spring Boot microservices into native
```
Comment (rhondaday) [inline docs-source/spring/content/development/setup.md:63]: "If ..., then you ..."

File: docs-source/spring/content/development/setup.md [hunk 26]
```diff
@@ -0,0 +1,128 @@
...
+images (which was officially supported from Spring Boot 3.0), you must use GraalVM, which can be
+downloaded [from here](https://www.graalvm.org/downloads/).
+
+## Maven
+
+You can use either Maven or Gradle to build your Spring Boot applications. If you prefer Maven, follow the steps in this section. If you prefer Gradle, refer to the next section.
```
Comment (rhondaday) [inline docs-source/spring/content/development/setup.md:69]: "If ..., then follow ..."

"If ..., then refer ...".

File: docs-source/spring/content/development/setup.md [hunk 27]
```diff
@@ -0,0 +1,128 @@
...
+
+Download Maven from the [Apache Maven website](https://maven.apache.org/download.cgi).
+
+Decompress the archive in your chosen location, for example your home directory, and then add it to your path:
```
Comment (rhondaday) [inline docs-source/spring/content/development/setup.md:73]: "Decompress ... location (for example, your home directory) and then ..."

File: docs-source/spring/content/development/setup.md [hunk 28]
```diff
@@ -0,0 +1,128 @@
...
+Default locale: en, platform encoding: UTF-8
+OS name: "linux", version: "5.10.102.1-microsoft-standard-wsl2", arch: "amd64", family: "unix"
+```
+
+## Gradle
+
+Download Gradle using the [instructions on the Gradle website](https://gradle.org/install/).
+Spring Boot is compatible with Gradle version 7.5 or later.
+
+Run the command below to verify Gradle was installed correctly:
```
Comment (rhondaday) [inline docs-source/spring/content/development/setup.md:95]: "Run the following command to verify that Gradle ..."

File: docs-source/spring/content/development/setup.md [hunk 29]
```diff
@@ -0,0 +1,128 @@
...
+Kotlin:       1.7.10
+Groovy:       3.0.13
+Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
+JVM:          17.0.3 (Oracle Corporation 17.0.3+8-LTS-111)
+OS:           Linux 5.10.102.1-microsoft-standard-WSL2 amd64
+```
+
+## Oracle Database in a container for local testing
+
+If you want to run an instance of Oracle Database locally for development and testing, Oracle
```
Comment (rhondaday) [inline docs-source/spring/content/development/setup.md:116]: If ..., then Oracle ..."

File: docs-source/spring/content/development/setup.md [hunk 30]
```diff
@@ -0,0 +1,128 @@
...
+recommends Oracle Database 23c Free.  You can start the database in a container with this
+command, you should specify a secure password:
```
Comment (rhondaday) [inline docs-source/spring/content/development/setup.md:118]: "... command specifying a secure password:"

File: docs-source/spring/content/development/setup.md [hunk 31]
```diff
@@ -0,0 +1,128 @@
...
+
+```bash
+docker run --name free23c -d \
+   -p 1521:1521 \
+   -e ORACLE_PWD=Welcome12345 \
+   container-registry.oracle.com/database/free:latest
+```
+
+**Note**: If you are using testcontainers, Oracle encourages you to consider this alternative
```
Comment (rhondaday) [inline docs-source/spring/content/development/setup.md:127]: "If you are using test containers, then Oracle ..."

File: docs-source/spring/content/development/_index.md [hunk 32]
```diff
@@ -0,0 +1,38 @@
...
+  Kubernetes Secrets and ConfigMaps for access to the Oracle Autonomous Database instance
+  associated with the backend.
+- An APISIX API Gateway that can be used to expose service endpoints outside the Kubernetes
+  cluster, to the public internet.  All standard APISIX features like traffic management, 
+  monitoring, authentication, and so on, are available for use.
+- Spring Eureka Service Registry for service discovery.  The API Gateway and monitoring
+  services are pre-configured to use this registry for service discovery.
+- Spring Config Server to serve externalized configuration information to applications.
+  This is configured to store the configuration data in the Oracle Autonomous Database
+  instance associated with the backend.
```
Comment (rhondaday) [inline docs-source/spring/content/development/_index.md:25]: "... backend:"

File: docs-source/spring/content/development/project.md [hunk 33]
```diff
@@ -0,0 +1,205 @@
...
+        </dependency>
+    </dependencies>
+</dependencyManagement>    
+```
+
+### Spring Application Configuration
+
+Oracle recommends the following configuration to configure the application access to
+use the built-in services, including the Spring Eureka Service Registry and the
+observability tools.
```
Comment (rhondaday) [inline docs-source/spring/content/development/project.md:70]: "... tools:"

File: docs-source/spring/content/development/project.md [hunk 34]
```diff
@@ -0,0 +1,205 @@
...
+      min-pool-size: 10
+      max-pool-size: 30
+```
+
+The variables in this configuration will be automatically injected to your deployment
+and pods when you use the Oracle Backend for Spring Boot and Microservices CLI to deploy your application. 
+
+#### Liquibase
+
+If you are using Liquibase to manage your database schema and data, you should
```
Comment (rhondaday) [inline docs-source/spring/content/development/project.md:141]: "If ..., then you ..."

File: docs-source/spring/content/development/setup.md [hunk 35]
```diff
@@ -0,0 +1,128 @@
...
+- macOS (11 or later recommended) on Intel or Apple silicon
+- Linux, for example Oracle Linux, Ubuntu, etc.
+
+The following tools are recommended for a development environment:
+
+- Integrated Development Environment, for example Visual Studio Code
+- Java Development Kit, for example Oracle, OpenJDK, or GraalVM 
+- Maven or Gradle for build and testing automation
+
+If you wish to test locally or offline, the following additional tools are recommended:
```
Comment (rhondaday) [inline docs-source/spring/content/development/setup.md:17]: "If ..., then the ..."

File: docs-source/spring/content/development/setup.md [hunk 36]
```diff
@@ -0,0 +1,128 @@
...
+You can install these by opening the extensions tab (Ctrl-Shift-X or equivalent) and using the
+search bar at the top to find and install them.
+
+## Java Development Kit
+
+Oracle recommends the [Java SE Development Kit](https://www.oracle.com/java/technologies/downloads/#java17)
+or [GraalVM](https://www.graalvm.org/downloads/#). Java 17 or 21 are recommended, note that Spring Boot
+3.0 requires at least Java 17.  If you want to use JVM Virtual Threads, then Java 21 is recommended.
+
+**Note**: If you are using Spring Boot 2.x, Oracle encourages you to use at least Java 17, unless
```
Comment (rhondaday) [inline docs-source/spring/content/development/setup.md:40]: "If ..., then Oracle ..."

File: docs-source/spring/content/development/_index.md [hunk 37]
```diff
@@ -0,0 +1,38 @@
...
+to use: 
+
+- An Oracle Autonomous Database instance in which applications can manage relational,
+  document, spatial, graph and other types of data, use Transactional Event Queues for
+  messaging and events using JMS, Kafka or REST interfaces, and even run ML models.
+- A Kubernetes cluster that applications can run in, with namespaces pre-configured with
+  Kubernetes Secrets and ConfigMaps for access to the Oracle Autonomous Database instance
+  associated with the backend.
+- An APISIX API Gateway that can be used to expose service endpoints outside the Kubernetes
+  cluster, to the public internet.  All standard APISIX features like traffic management,
```
Comment (rhondaday) [inline docs-source/spring/content/development/_index.md:19]: use "Apache APISIX"

File: docs-source/spring/content/development/project.md [hunk 38]
```diff
@@ -0,0 +1,205 @@
+---
+title: "Project Structure"
+---
+
+To take advantage of the built-in platform services, Oracle recommends that you consider
```
Comment (rhondaday) [inline docs-source/spring/content/development/project.md:5]: "... Oracle recommends using the ..."

File: docs-source/spring/content/development/project.md [hunk 39]
```diff
@@ -0,0 +1,205 @@
...
+
+Recommended versions:
+
+* Spring Boot 3.1.5
+* Spring Cloud 2022.0.4
+* Java 17 or 21
+
+### Dependencies
+
+Oracle recommends you add the following dependencies to your application so that it
```
Comment (rhondaday) [inline docs-source/spring/content/development/project.md:16]: "Oracle recommends adding the following ..."

File: docs-source/spring/content/development/project.md [hunk 40]
```diff
@@ -0,0 +1,205 @@
...
+<dependencies>
+    <dependency>
+        <groupId>org.liquibase</groupId>
+        <artifactId>liquibase-core</artifactId>
+        <version>${liquibase.version}</version>
+    </dependency>
+</dependencies>
+```
+
+The following configuration should be added to your Spring application configuration:
```
Comment (rhondaday) [inline docs-source/spring/content/development/project.md:158]: Change to "Add the following configuration to your Spring application configuration:"

File: docs-source/spring/content/development/project.md [hunk 41]
```diff
@@ -0,0 +1,205 @@
...
+
+```xml
+<dependency>
+     <groupId>com.oracle.microtx.lra</groupId>
+     <artifactId>microtx-lra-spring-boot-starter-3x</artifactId>
+     <version>23.4.1</version>
+</dependency>
+```
+
+The following configuration should be added to your Spring application configuration:
```
Comment (rhondaday) [inline docs-source/spring/content/development/project.md:192]: Change to "Add the following configuration to your Spring application configuration:"

File: docs-source/spring/content/development/setup.md [hunk 42]
```diff
@@ -0,0 +1,128 @@
...
+[this permalink](https://download.oracle.com/java/17/latest/jdk-17_linux-x64_bin.tar.gz).
+
+Decompress the archive in your chosen location, for example your home directory and then add it to your path:
+
+```bash
+export JAVA_HOME=$HOME/jdk-17.0.3
+export PATH=$JAVA_HOME/bin:$PATH
+```
+
+You can verify it is installed with this command:
```
Comment (rhondaday) [inline docs-source/spring/content/development/setup.md:54]: Change to "Use the following command to verify that it is installed:"

File: docs-source/spring/content/development/setup.md [hunk 43]
```diff
@@ -0,0 +1,128 @@
...
+
+Download Maven from the [Apache Maven website](https://maven.apache.org/download.cgi).
+
+Decompress the archive in your chosen location, for example your home directory, and then add it to your path:
+
+```bash
+$ export PATH=$HOME/apache-maven-3.8.6/bin:$PATH
+```
+
+You can verify it is installed with this command (note that your version may give slightly different output):
```
Comment (rhondaday) [inline docs-source/spring/content/development/setup.md:79]: Change to "Use the following command to verify it is installed (note ..."

**Review by rhondaday — COMMENTED** [review]
Several comments to incorporate.

**Review by andytael — APPROVED** [review]

**Pull request merged**